### PR TITLE
Newline trim fix

### PR DESCRIPTION
**Pull request by guggero**
This PR fixes an issue where trailing newline characters (0x0d and 0x0a) were trimmed from macaroon files.

### CODE REVIEW

File: cmd_gen_password.go [hunk 1]
```diff
@@ -73,7 +73,7 @@ func (x *genPasswordCommand) Execute(_ []string) error {
 		}
 	}
 
-	fmt.Printf("%s\n", passwordString)
```
Comment (Roasbeef) [inline cmd_gen_password.go:76]: This is just a defense in depth fix, and the seeds actually weren't affected by this issue right? Or I guess maybe eventually a deployed seed would, as it's a random event that the characters line up

File: k8s.go [hunk 2]
```diff
@@ -93,13 +93,9 @@ func readK8s(opts *k8sSecretOptions) (string, *jsonK8sObject, error) {
 			opts.SecretKeyName)
 	}
 
-	// Remove any newlines at the end of the file. We won't ever write a
-	// newline ourselves but maybe the file was provisioned by another
-	// process or user.
-	content := stripNewline(string(secret.Data[opts.SecretKeyName]))
```
Comment (Roasbeef) [inline k8s.go:99]: One way we can programmatically confirm/harden the fix would be to extra this routine into a simple function and add a test (based on a similar instance as we saw in prod) to make sure future changes here don't regress.

**Review by Roasbeef — CHANGES_REQUESTED** [review]
Easy fix, only comment is to extract the routine to add a basic test case to make sure we don't regress to what incident that happened on prod after deployment.

File: cmd_gen_password.go (hunk 1 above)
Comment (guggero) [reply cmd_gen_password.go:76]: Just to be a bit more consistent. A tool that is sensitive to newline characters in certain things should also be careful about not writing them by default. Also, we have comments like "lndinit doesn't add newlines itself" in some places. So I think this is just best practice.

**Review by Roasbeef — APPROVED** [review]
LGTM 🦜

File: k8s_test.go [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+)
+
+// TestSecretToString makes sure that a raw secret can be turned into a string
+// correctly.
+func TestSecretToString(t *testing.T) {
+	testCases := []struct {
+		name      string
+		input     []byte
+		base64    bool
+		expectErr bool
```
Comment (bhandras) [inline k8s_test.go:26]: I'd also add at least one testcase to cover `expectErr == true`

**Review by bhandras — APPROVED** [review]

**Pull request merged**